.github/workflows/ci.yaml

```diff
@@ -0,0 +1,81 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches:
+    - main
+    - track/**
+
+# kill all previous jobs in the same branch if new commits are pushed
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  quality-checks:
+    name: Quality Checks
+    strategy:
+      matrix:
+        charm:
+          - kfp-api
+          - kfp-persistence
+          - kfp-profile-controller
+          - kfp-schedwf
+          - kfp-ui
+          - kfp-viewer
+          - kfp-viz
+    uses: canonical/charmed-kubeflow-workflows/.github/workflows/_quality-checks.yaml@all
```

Comment:
Why do we have an `all` branch?

Reply:
all branch is in canonical/charmed-kubeflow-workflows

once we finalize, I think we would need to switch to tags and V1, V2, etc

currently, I incorporate feedback directly in the branch and do not need to reapply tags